sysl2/sysl/grammar/SyslParser.g4

```diff
@@ -42,6 +42,11 @@ table   :   SYSL_COMMENT*
             (TABLE | TYPE)
             name_str attribs_or_modifiers? COLON ( WHATEVER | INDENT (SYSL_COMMENT | field | annotation | inplace_table | WHATEVER )+ DEDENT)
         ;
+union   :   SYSL_COMMENT*
+            UNION
+            name_str attribs_or_modifiers? COLON ( WHATEVER | INDENT (SYSL_COMMENT | type_ref | annotation | inplace_table | WHATEVER )+ DEDENT)
```

Comment:
can you use `user_defined_type` instead of defining a new rule?

Reply:
Yes, but it breaks `EnterUser_defined_type` because it is now triggered outside the `types` context. I've made the change and added a short-circuit return to avoid the problem. All tests pass, so it can't be too disastrous, right?